### PR TITLE
Coalesce hosting model coverage

### LINKED ISSUE #12099: Assess location of hosting model coverage
The hosting model coverage is primarily found in ...

* [Servers](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/?view=aspnetcore-2.2&tabs=windows#hosting-models)
* [ANCM](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/aspnet-core-module?view=aspnetcore-2.2#hosting-models)

There's a little coverage in the [IIS Overview](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/?view=aspnetcore-2.2#enable-the-iisintegration-components). Although the coverage is relevant to Azure Apps, it's only indirectly linked to the Azure Apps topic.

* Assess the coverage and cross-linking.
* Look for opportunities to coalesce the coverage: Try to keep all of the implementation coverage in one location and link to it from other locations.

Check one of the early PRs for this: I think there was a different organization on the initial commit that was changed on a later commit. https://github.com/aspnet/AspNetCore.Docs/pull/10002

cc: @shirhatti

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fe69b3a9-896f-e189-0c2c-f87ff735df9c
* Version Independent ID: 97c8549b-66ad-0a23-1971-784fb923042a
* Content: [ASP.NET Core Module](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/aspnet-core-module?view=aspnetcore-2.2)
* Content Source: [aspnetcore/host-and-deploy/aspnet-core-module.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/host-and-deploy/aspnet-core-module.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-hostdeploy**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #12099 

* [Internal Review Topic (Servers node overview, Hosting models section)](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/index?view=aspnetcore-2.2&branch=pr-en-us-12573#hosting-models)
* [Internal Review Topic (IIS Overview; Hosting models section)](https://review.docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/index?view=aspnetcore-2.2&branch=pr-en-us-12573#hosting-models)

The live coverage today is spread among:

* [Web server implementations (*Servers node overview*)](https://docs.microsoft.com/aspnet/core/fundamentals/servers/#hosting-models)
* [IIS Overview](https://docs.microsoft.com/aspnet/core/host-and-deploy/iis/#enable-the-iisintegration-components)
* [ANCM](https://docs.microsoft.com/aspnet/core/host-and-deploy/aspnet-core-module#hosting-models)

This seeks to coalesce conceptual coverage under the *IIS Overview*. The *ANCM topic* continues to maintain the implementation guidance.

The *Servers node overview* is left with a quick paragraph on each and cross-links down to the *IIS Overview* and ANCM topic for the details.

[EDIT] Another way that occurred to me (if IIS is just *waaay toooo long now*) is to create a new topic under the IIS node for hosting model coverage ... then cross-link to it from these various locations. Let me know if you'd prefer that approach.

### CODE REVIEW

File: aspnetcore/fundamentals/servers/index.md [hunk 1]
```diff
@@ -20,56 +21,23 @@ An ASP.NET Core app runs with an in-process HTTP server implementation. The serv
...
 When using [IIS](/iis/get-started/introduction-to-iis/introduction-to-iis-architecture) or [IIS Express](/iis/extensions/introduction-to-iis-express/iis-express-overview), the app either runs:
 
-* In the same process as the IIS worker process (the [in-process hosting model](#in-process-hosting-model)) with the [IIS HTTP Server](#iis-http-server). *In-process* is the recommended configuration.
-* In a process separate from the IIS worker process (the [out-of-process hosting model](#out-of-process-hosting-model)) with the [Kestrel server](#kestrel).
+* In the same process as the IIS worker process (the [in-process hosting model](#hosting-models)) with the [IIS HTTP Server](#iis-http-server). *In-process* is the recommended configuration.
+* In a process separate from the IIS worker process (the [out-of-process hosting model](#hosting-models)) with the [Kestrel server](#kestrel).
 
 The [ASP.NET Core Module](xref:host-and-deploy/aspnet-core-module) is a native IIS module that handles native IIS requests between IIS and the in-process IIS HTTP Server or Kestrel. For more information, see <xref:host-and-deploy/aspnet-core-module>.
 
 ## Hosting models
```
Comment (jkotalik) [inline aspnetcore/fundamentals/servers/index.md]: Reorder this to have kestrel first in the list of servers (as it's the only one that is cross plat).

File: aspnetcore/fundamentals/servers/index.md [hunk 2]
```diff
@@ -174,7 +124,7 @@ For information on how to use Apache on Linux as a reverse proxy server for Kest
 
 ## IIS HTTP Server
```
Comment (jkotalik) [inline aspnetcore/fundamentals/servers/index.md]: Delete the IIS HTTP Server, inprocess and out of process hosting cover it.

**Comment by guardrex** [comment]
@shirhatti We'll hold this here until the "PU"-label issues are addressed. It's probably best if you look :eyes: and confirm that this is what you were hoping for to address the feedback that you received.

**Comment by guardrex** [comment]
Hi @shirhatti Ur doing the doc-a-thon today? If so, is this on your radar?

**Comment by guardrex** [comment]
@Rick-Anderson I recommend we go ahead with this. We can react to PU requests later.

**Pull request merged**